src/main/java/seedu/planner/model/util/ModuleUtil.java

```diff
@@ -37,13 +39,23 @@ private static boolean hasNotTakenModule(List<Module> modulesTaken, Module modul
      * @param moduleToCheck The {@code Module} to be checked.
      * @return True if all the prerequisites have been taken.
      */
-    private static boolean hasFulfilledAllPrerequisites(List<Module> modulesTaken, Module moduleToCheck) {
+    public static boolean hasFulfilledAllPrerequisites(List<Module> modulesTaken, Module moduleToCheck) {
         List<ModuleInfo> prerequisites = moduleToCheck.getPrerequisites();
+        List<List<ModuleInfo>> groupedByEquivalence = groupModuleInfosByEquivalence(prerequisites);
 
-        for (ModuleInfo p: prerequisites) {
-            Module m = new Module(p.getCode());
+        for (List<ModuleInfo> equivalence : groupedByEquivalence) {
+            boolean oneIsNotContained = true;
```

Comment:
Change to perhaps `isOneNotContained` to follow boolean variable naming standard